homeassistant/components/vallox/fan.py

```diff
@@ -0,0 +1,167 @@
+"""Support for the Vallox ventilation unit fan."""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.fan import FanEntity
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import (
+    DOMAIN, METRIC_KEY_MODE, METRIC_KEY_PROFILE_FAN_SPEED_AWAY,
+    METRIC_KEY_PROFILE_FAN_SPEED_BOOST, METRIC_KEY_PROFILE_FAN_SPEED_HOME,
+    SIGNAL_VALLOX_STATE_UPDATE)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Device attributes
+ATTR_PROFILE_FAN_SPEED_HOME = {
+    'description': 'fan_speed_home',
+    'metric_key': METRIC_KEY_PROFILE_FAN_SPEED_HOME
+}
+ATTR_PROFILE_FAN_SPEED_AWAY = {
+    'description': 'fan_speed_away',
+    'metric_key': METRIC_KEY_PROFILE_FAN_SPEED_AWAY
+}
+ATTR_PROFILE_FAN_SPEED_BOOST = {
+    'description': 'fan_speed_boost',
+    'metric_key': METRIC_KEY_PROFILE_FAN_SPEED_BOOST
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the fan device."""
+    try:
+        device = ValloxFan(hass.data[DOMAIN]['name'],
+                           hass.data[DOMAIN]['client'],
+                           hass.data[DOMAIN]['state_proxy'])
+    except KeyError:
```

Comment:
Remove this. Instead add a guard clause at the top of the function body that checks if discovery_info is None and return if so.